superset-frontend/spec/javascripts/explore/components/withAsyncVerification_spec.tsx

```diff
@@ -0,0 +1,142 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { mount, ReactWrapper } from 'enzyme';
+import { act } from 'react-dom/test-utils';
+
+import MetricsControl from 'src/explore/components/controls/MetricsControl';
+import withAsyncVerification, {
+  ControlPropsWithExtras,
+  WithAsyncVerificationOptions,
+} from 'src/explore/components/controls/withAsyncVerification';
+import { ExtraControlProps } from '@superset-ui/chart-controls';
+
+const VALID_METRIC = {
+  metric_name: 'sum__value',
+  expression: 'SUM(energy_usage.value)',
+};
+
+const mockSetControlValue = jest.fn();
+
+const defaultProps = {
+  name: 'metrics',
+  label: 'Metrics',
+  value: undefined,
+  multi: true,
+  needAsyncVerification: true,
+  actions: { setControlValue: mockSetControlValue },
```

Comment:
`setControlValue` is required (as is used in `FilterBox`), all other actions are optional.